File: product-flow-client/src/types/product.ts
export interface Product {
    id: number;
    name: string;
    productCode: string;
    price: number;
    quantity: number;
}

export interface ProductCreateDto {
    name: string;
    productCode: string;
    price: number;
    quantity: number;
}

export interface ProductUpdateDto {
    id: number;
    name: string;
    productCode: string;
    price: number;
    quantity: number;
}

File: ProductFlowServer/ProductFlowServer.Entities/Concrete/Product.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ProductFlowServer.Entities.Concrete
{
    public sealed class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string ProductCode { get; set; }
        public decimal Price { get; set; }
        public int Quantity { get; set; }
        public bool IsActive { get; set; }
        public DateTime? CreateDate { get; set; }
        public DateTime? UpdateDate { get; set; }
    }
}
